### DISCUSSION TITLE
Automerge issues

### DISCUSSION THREAD

**Discussion by mfinegold-cambia** [renovatebot/renovate#9945] [Request Help]
I'm using Gitlab and would like to group all devDependencies into 1 MR/branch, and then have that branch automerged.  When I use the below packageRules, it creates the devDependencies MR/branch but doesn't automerge afterwards.   I tried adding  requiredStatus: null  below the automerge field, and that doesn't seem to create the MR/branch or do anything.  My Gitlab project doesn't have any restrictions on which users can merge to master or requiring an MR first. Any idea?

  packageRules: [
    {
      matchDepTypes: [ "devDependencies"],
      groupName: "devDependencies",
      automerge: true
    },
    {
      matchPackageNames: [ "aws-sdk" ],
      matchDepTypes: [ "dependencies" ],
      matchUpdateTypes: [ "minor", "patch" ], 
      enabled: false
    },
    {
      matchUpdateTypes: [ "minor", "patch" ],
      matchDepTypes: [ "dependencies" ],
      groupName: "Minor and Patch Versions",
      excludePackagePrefixes: [ "@chp-cps" ]
    }
  ],

**Comment by viceice** [comment]
Please provide debug logs. They should show you what's going on. Or why it won't automerge.


⚠️ Automerge without CI checks is not recommend, it can silently break your default branch. So be warned.

**Reply by mfinegold-cambia** [reply]
Thanks for your help.  The logs are below.  I see a "PR is configured to automerge" entry and a "PR is not ready for merge" entry.  What is it looking for to determine if the PR is ready to merge?   When the devdependencies branch was created, my pipeline build job succeeded.  The MR shows that automerge is Enabled.  Am I missing a config setting to have this automerge after the gitlab MR/branch is created?


{"name":"renovate","hostname":"e9a6799eccf8","pid":24249,"level":20,"logContext":"2RtxEVEdx","repository":"chp-consumer-profile-services/chp-cps-vendor-callback-adapter","branch":"dependency-update/devdependencies","branchStatus":"yellow","branchName":"dependency-update/devdependencies","msg":"getBranchStatus() result","time":"2021-05-10T18:36:41.163Z","v":0}
{"name":"renovate","hostname":"e9a6799eccf8","pid":24249,"level":20,"logContext":"2RtxEVEdx","repository":"chp-consumer-profile-services/chp-cps-vendor-callback-adapter","branch":"dependency-update/devdependencies","msg":"Skipping assignees and reviewers as automerge=true","time":"2021-05-10T18:36:41.163Z","v":0}
{"name":"renovate","hostname":"e9a6799eccf8","pid":24249,"level":20,"logContext":"2RtxEVEdx","repository":"chp-consumer-profile-services/chp-cps-vendor-callback-adapter","branch":"dependency-update/devdependencies","msg":"Created Merge Request #44","time":"2021-05-10T18:36:41.163Z","v":0}
**{"name":"renovate","hostname":"e9a6799eccf8","pid":24249,"level":20,"logContext":"2RtxEVEdx","repository":"chp-consumer-profile-services/chp-cps-vendor-callback-adapter","branch":"dependency-update/devdependencies","msg":"PR is configured for automerge","time":"2021-05-10T18:36:41.163Z","v":0}**
**{"name":"renovate","hostname":"e9a6799eccf8","pid":24249,"level":20,"logContext":"2RtxEVEdx","repository":"chp-consumer-profile-services/chp-cps-vendor-callback-adapter","branch":"dependency-update/devdependencies","msg":"PR is not ready for merge","time":"2021-05-10T18:36:41.163Z","v":0}**
{"name":"renovate","hostname":"e9a6799eccf8","pid":24249,"level":20,"logContext":"2RtxEVEdx","repository":"chp-consumer-profile-services/chp-cps-vendor-callback-adapter","msg":"getBranchPr(dependency-update/devdependencies)","time":"2021-05-10T18:36:41.178Z","v":0}
{"name":"renovate","hostname":"e9a6799eccf8","pid":24249,"level":20,"logContext":"2RtxEVEdx","repository":"chp-consumer-profile-services/chp-cps-vendor-callback-adapter","msg":"findPr(dependency-update/devdependencies, undefined, open)","time":"2021-05-10T18:36:41.178Z","v":0}
{"name":"renovate","hostname":"e9a6799eccf8","pid":24249,"level":20,"logContext":"2RtxEVEdx","repository":"chp-consumer-profile-services/chp-cps-vendor-callback-adapter","msg":"Removing any stale branches","time":"2021-05-10T18:36:41.179Z","v":0}
{"name":"renovate","hostname":"e9a6799eccf8","pid":24249,"level":20,"logContext":"2RtxEVEdx","repository":"chp-consumer-profile-services/chp-cps-vendor-callback-adapter","msg":"config.repoIsOnboarded=true","time":"2021-05-10T18:36:41.179Z","v":0}

**Reply by rarkins** [reply]
Renovate won't automerge in the same run as it creates the MR. Do you have tests and if so then wouldn't you want them to run first?

**Reply by mfinegold-cambia** [reply]
Yes, the way my pipeline is set up is that when a new branch is created/pushed, I have 2 jobs that are run automatically :  Build and Test.   And then I have a subsequent manual job (  when: manual  in gitlab ci yaml ) that deploys it to an environment.  It looks like Renovate is automerging only after I trigger my manual deploy job.  But I'd like it to auto-merge after the Test job succeeds.  What is Renovate looking at in gitlab to know to automerge?

**Reply by rarkins** [reply]
You need to capture a log when Renovate runs and when the branch has passed all tests. Renovate will check the status checks for the commit.
